### PR TITLE
Photon: Ensure VIP site are correctly identified within isVIP.

### PR DESCRIPTION
**Pull request by coder-karen**
Fixes https://github.com/Automattic/jpop-issues/issues/9291

## Proposed changes:

* Th issue this PR fixes is that Image Compare block images were being served by Photon if a VIP site, when they should not have been.
* The primary issue is that the Image Compare block is calling `photonizedImgProps` without ensuring the `jetpack_plan` variable had been set, leading to images being photonized if no Tiled Gallery block existed on the page.
* This PR ensures that the plan check within `isVIP()` is correct, by instead including the plan value in `Jetpack_Editor_Initial_State`. Previously, the plan check (creating a global `jetpack_plan` variable) was made in the render function within `tiled-gallery.php`, which would only run if there was a Tiled Gallery on the post / page. 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/jpop-issues/issues/9291
p1729613258801529/1729549827.901559-slack-CDD9LQRSN
p1729607065573149-slack-CDD9LQRSN

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

VIP:

If you have access to a VIP test site:

* This would need to be tested on a VIP site, using the Image Compare block, to ensure images are not served via Photon. Testing would include adding a Tiled Gallery block in one post and Image Compare in another, in the current Jetpack version. Then apply the changes, and ensure there are no block validation issues. Then ensure newly added Image Compare blocks are not serving images via Photon.

Code Review:

* As it is difficult to get a testing environment for VIP, a code review would also be helpful here.
* We should also ensure that the Tiled Gallery and Image Compare block on non-VIP sites continue to work in the same manner as before, with no errors.
* We can also test that there are no issues if upgrading Jetpack from an older version where a Tiled Gallery was added, to a newer version. If doing this, download JP version 13.5 to test with - https://downloads.wordpress.org/plugin/jetpack.13.5.zip - as the following version (13.6) is the most recent version with a deprecation. The intention is to make sure there are no errors related to the `jetpack_plan` variable.
* Note if doing so, you will likely encounter block validation errors, but this is related to a different issue: p8oabR-1At-p2#comment-8295 - we're just looking for issues related to the `jetpack_plan` variable.

### CODE REVIEW

**Comment by sergeymitr** [comment]
The code looks good and non-VIP sites work just fine.
Asked VIP for help with testing here: p1729607065573149-slack-CDD9LQRSN

File: projects/plugins/jetpack/extensions/blocks/tiled-gallery/utils/index.js [hunk 1]
```diff
@@ -112,9 +117,14 @@ export function photonizedImgProps( img, galleryAtts = {} ) {
...
-	if ( typeof jetpack_plan !== 'undefined' && jetpack_plan.data === 'vip' ) {
-		return true;
+	// Use `jetpackPlanFromState` if available, otherwise fall back to `jetpack_plan` defined within the render function in tiled-gallery.php.
+	let jetpackPlan;
+	if ( typeof jetpackPlanFromState !== 'undefined' ) {
+		jetpackPlan = jetpackPlanFromState;
+	} else if ( typeof jetpack_plan !== 'undefined' ) {
+		jetpackPlan = jetpack_plan;
 	}
+	return jetpackPlan && jetpackPlan.data === 'vip';
```
Comment (sergeymitr) [inline projects/plugins/jetpack/extensions/blocks/tiled-gallery/utils/index.js]: Do you think we should make sure `data` property exists?
```suggestion
	return jetpackPlan && jetpackPlan?.data === 'vip';
```

**Review by sergeymitr — DISMISSED** [review]

File: projects/plugins/jetpack/extensions/blocks/tiled-gallery/utils/index.js (hunk 1 above)
Comment (coder-karen) [reply projects/plugins/jetpack/extensions/blocks/tiled-gallery/utils/index.js]: Good idea.

**Comment by coder-karen** [comment]
Noting here that we're currently waiting for VIP testing.

**Comment by coder-karen** [comment]
Opening this up for review / code review again after more changes have been added, based on VIP testing. See p1734517123407559/1729607065.573149-slack-CDD9LQRSN.

**Review by darssen — APPROVED** [review]
The new code changes look good and I don't see any issues with Image Block and Tiled Gallery blocks.
I haven't tested in VIP environment though.

**Pull request merged**